### PR TITLE
NEW : manage several type of dangerous goods for a same parcel

### PR DESCRIPTION
**Pull request by dolibarr95**
Manage several type of dangerous goods for a same parcel 
Before : dangerous_goods : smallint (so only 1 type can be defined eg 3) 
After dangerous_goods : varchar so can be a list of numbers eg 3,5,8

### CODE REVIEW

File: htdocs/install/mysql/tables/llx_expedition_package.sql [hunk 1]
```diff
@@ -12,7 +12,7 @@ create table llx_expedition_package
   size_units        integer,	       -- unit of all sizes (height, width, depth)
   weight            float,	       -- weight
   weight_units      integer,	       -- unit of weight
-  dangerous_goods   smallint          DEFAULT 0, -- 0 = no dangerous goods or 1 = Explosives, 2 = Flammable Gases, 3 = Flammable Liquids, 4 = Flammable solids, 5 = Oxidizing, 6 = Toxic & Infectious, 7 = Radioactive, 8 = Corrosives, 9 = Miscellaneous (see https://en.wikipedia.org/wiki/Dangerous_goods). I'm not sure if just register 0 (no) or 1 (yes) is enough.
+  dangerous_goods   varchar(20), --(comma separated list eg : 3,5,7) 0 = no dangerous goods or 1 = Explosives, 2 = Flammable Gases, 3 = Flammable Liquids, 4 = Flammable solids, 5 = Oxidizing, 6 = Toxic & Infectious, 7 = Radioactive, 8 = Corrosives, 9 = Miscellaneous (see https://en.wikipedia.org/wiki/Dangerous_goods). I'm not sure if just register 0 (no) or 1 (yes) is enough.
```
Comment (eldy) [inline htdocs/install/mysql/tables/llx_expedition_package.sql:15]: If you change into sql file, you must alos add the migration instruction into the migration file 20.0.0-21.0.0.sql

Comment (dolibarr95) [reply htdocs/install/mysql/tables/llx_expedition_package.sql:15]: Hello
ok, thanks for the advice I'll do this.

**Comment by dolibarr95** [comment]
Done ✔️

File: htdocs/install/mysql/tables/llx_expedition_package.sql [hunk 2]
```diff
@@ -12,7 +12,7 @@ create table llx_expedition_package
   size_units        integer,	       -- unit of all sizes (height, width, depth)
   weight            float,	       -- weight
   weight_units      integer,	       -- unit of weight
-  dangerous_goods   smallint          DEFAULT 0, -- 0 = no dangerous goods or 1 = Explosives, 2 = Flammable Gases, 3 = Flammable Liquids, 4 = Flammable solids, 5 = Oxidizing, 6 = Toxic & Infectious, 7 = Radioactive, 8 = Corrosives, 9 = Miscellaneous (see https://en.wikipedia.org/wiki/Dangerous_goods). I'm not sure if just register 0 (no) or 1 (yes) is enough.
+  dangerous_goods   varchar(20) DEFAULT '0', --(comma separated list eg : 3,5,7) 0 = no dangerous goods or 1 = Explosives, 2 = Flammable Gases, 3 = Flammable Liquids, 4 = Flammable solids, 5 = Oxidizing, 6 = Toxic & Infectious, 7 = Radioactive, 8 = Corrosives, 9 = Miscellaneous (see https://en.wikipedia.org/wiki/Dangerous_goods). I'm not sure if just register 0 (no) or 1 (yes) is enough.
```
Comment (fappels) [inline htdocs/install/mysql/tables/llx_expedition_package.sql:15]: space missing after --

Comment (dolibarr95) [reply htdocs/install/mysql/tables/llx_expedition_package.sql]: tks for your help

**Comment by dolibarr95** [comment]
@eldy done ✔️

File: htdocs/install/mysql/migration/20.0.0-21.0.0.sql [hunk 3]
```diff
@@ -47,3 +47,6 @@ ALTER TABLE llx_contrat ADD COLUMN localtax2 double(24,8) DEFAULT 0;
 ALTER TABLE llx_contrat ADD COLUMN revenuestamp double(24,8) DEFAULT 0;
 ALTER TABLE llx_contrat ADD COLUMN total_ht double(24,8) DEFAULT 0;
 ALTER TABLE llx_contrat ADD COLUMN total_ttc double(24,8) DEFAULT 0;
+
+ALTER TABLE llx_expedition_package CHANGE dangerous_goods dangerous_goods VARCHAR(20) NOT NULL DEFAULT '0';
```
Comment (eldy) [inline htdocs/install/mysql/migration/20.0.0-21.0.0.sql:51]: See in head of file the comment
-- To change type of field: ALTER TABLE llx_table MODIFY COLUMN name varchar(60);

**Comment by dolibarr95** [comment]
Hello
@eldy and @fappels tks again for your help.

**Pull request merged**